src/main/java/com/minecolonies/coremod/client/gui/WindowListRecipes.java

```diff
@@ -147,6 +157,7 @@ public void updateElement(final int index, @NotNull final Pane rowPane)
     public void onUpdate()
     {
         updateRecipes();
+        recipeStatus.setLabelText(LanguageHandler.format("com.minecolonies.coremod.gui.workerhuts.recipestatus",building.getRecipes().size(),AbstractBuildingCrafter.buildingMaxRecipes(building.getBuildingLevel())));
```

Comment:
You should not use a static accessor here for buildingMaxRecipe. There is a possibility not every building has the same number of recipes, so you need it to be overridable.  A good example of this is the lumberjack, where the recipe count and max recipes are always the same.

Reply:
And it needs to take the research for more recipe slots into account.

Reply:
Also constants!
All translationkeys go in TranslationKeys